src/Workspaces/Core/Portable/LanguageServices/SyntaxFactsService/AbstractSyntaxFactsService.cs

```diff
@@ -95,10 +99,144 @@ protected static void FreeAliasMap(Dictionary<string, string> aliasMap)
         }
 
         protected static Dictionary<string, string> AllocateAliasMap()
+            => s_aliasMapPool.Allocate();
+
+        public bool IsOnSingleLine(SyntaxNode node, bool fullSpan)
```

Comment:
How would it get line numbers for an arbitrary node?  The node may be synthetic, in which case is does not belong to a tree.  This would cause us to have to allocate a tree, and then create an entire source text to answer that question.